File: src/model_v4/entities/SBS_LIZTYP.ts
import { Column, Entity, OneToMany, PrimaryColumn } from "typeorm";
import { SBS_SW } from "./SBS_SW";


@Entity("SBS_LIZTYP",{schema:"sbsdb"})
export class SBS_LIZTYP {

    @PrimaryColumn("bigint",{ 
        nullable:false,
        primary:true,
        name:"LIZTYP_INDEX"
        })
    LIZTYP_INDEX:string;
        

    @Column("bigint",{ 
        nullable:false,
        name:"ABRECHNUNG"
        })
    ABRECHNUNG:string;
        

    @Column("varchar",{ 
        nullable:false,
        length:50,
        name:"LIZENZIERUNG"
        })
    LIZENZIERUNG:string;
        

   
    @OneToMany(type=>SBS_SW, SBS_SW=>SBS_SW.lIZTYP_INDEX,{ onDelete: 'RESTRICT' ,onUpdate: 'RESTRICT' })
    sBS_SWs:SBS_SW[];
    
}

File: SbsdbServer/sbsdbv4/model/SbsLiztyp.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hb.SbsdbServer.sbsdbv4.model {
    [Table("SBS_LIZTYP")]
    public class SbsLiztyp {
        public SbsLiztyp() {
            SbsSw = new HashSet<SbsSw>();
        }

        [Column("LIZTYP_INDEX", TypeName = "bigint(20)")]
        public long LiztypIndex { get; set; }

        [Column("ABRECHNUNG", TypeName = "bigint(20)")]
        public long Abrechnung { get; set; }

        [Required]
        [Column("LIZENZIERUNG", TypeName = "varchar(50)")]
        public string Lizenzierung { get; set; }

        [InverseProperty("LiztypIndexNavigation")]
        public virtual ICollection<SbsSw> SbsSw { get; set; }
    }
}
